aten/src/ATen/native/mps/operations/UnaryOps.mm

```diff
@@ -35,7 +34,14 @@ void unary_op(const Tensor& self, const Tensor& output, std::string op_name, Una
           MPSGraph* mpsGraph = make_mps_graph();
           newCachedGraph = new CachedGraph(mpsGraph);
           newCachedGraph->inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, self);
-          newCachedGraph->outputTensor = unaryBlock(mpsGraph, newCachedGraph->inputTensor);
+          MPSGraphTensor* castTensor = newCachedGraph->inputTensor;
+          // Integer input must be cast to float if output is float
+          if (isIntegralType(self.scalar_type()) && !isIntegralType(output.scalar_type())) {
```

Comment:
Here we should be explicit about the isFloatingType for output. There are all the bfloat, complex and other types which can go down this path. I think it will be better use floating type in condition